mycroft/gui/extensions.py

```diff
@@ -0,0 +1,243 @@
+import threading
+from mycroft.configuration import Configuration
+from mycroft.messagebus import Message
+from mycroft.util.log import LOG
+from mycroft.api import is_paired
+from ovos_utils.system import system_reboot, system_shutdown, ssh_enable, ssh_disable
+
+from json_database import JsonStorageXDG
+from mycroft.gui.homescreen import HomescreenManager
+
+
+class ExtensionsManager():
+    def __init__(self, name, bus, gui):
+        """ Constructor for the Extension Manager. The Extension Manager is responsible for
+        managing the extensions that define additional GUI behaviours for specific platforms.
+
+        Args:
+            name: Name of the extension manager
+            bus: MessageBus instance
+            gui: GUI instance
+        """
+
+        self.name = name
+        self.bus = bus
+        self.gui = gui
+        config = Configuration.get()
+        enclosure_config = config.get("enclosure")
+        self.active_extension = enclosure_config.get("extension", "Generic")
+
+        # ToDo: Add Exclusive Support For "Desktop", "Mobile" Extensions
+        self.supported_extensions = ["SmartSpeaker", "Bigscreen", "Generic"]
```

Comment:
can you make this a enum at the top of the file?  to make the enum also be a regular string subclass str first

```python
class GUIExtensions(str, Enum):
     SMARTSPEAKER = "SmartSpeaker"
     BIGSCREEN = "Bigscreen"
     GENERIC = "Generic"
```


can the strings also be lower case for consistency with all plugin modules?

Reply:
ignore this comment, this is good for future pluginification instead